api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxCurrCohortQueries.java

```diff
@@ -40,6 +38,11 @@
 	        + "and obs.obs_datetime = (select max(encounter.encounter_datetime) from encounter "
 	        + "where encounter.encounter_type in (%s) and encounter.patient_id = obs.person_id and encounter.location_id = obs.location_id and encounter.voided = false and encounter.encounter_datetime <= :onOrBefore) ";
 	
+	private static final String EXCLUDE_PATIENTS_ENRROLLED_BUT_WITHOUT_CONSULTATION = "select distinct(person.person_id) from patient_program "
```

Comment:
@Guimino lets improve this query by including the check that patients have an encounter using a left join - it is more efficient and a bit cleaner way to write the query instead of using an inner select on all encounters.

Something like this:
```
select distinct(person.person_id) 
from patient_program 
	join person on person.person_id = patient_program.patient_id
	left join encounter on (encounter.patient_id=patient_program.patient_id and encounter.location_id = :location)
where patient_program.location_id = :location
	and patient_program.program_id = %s 
	and person.voided = 0 	
	and encounter.patient_id is null
```

Reply:
ok @nthfloor

Reply:
@nthfloor , this sql returns the same results as the previous query  of exclusion and this last is more performatic. I will change to use this for the exclusion